engine/src/flutter/impeller/entity/contents/content_context.h

```diff
@@ -547,16 +560,29 @@ class ContentContext {
       return nullptr;
     }
 
-    PipelineHandleT* GetDefault() const {
+    bool IsDefault(const ContentContextOptions& opts) {
+      return default_options_.has_value() &&
+             opts.ToKey() == default_options_.value().ToKey();
+    }
+
+    PipelineHandleT* GetDefault(const Context& context) {
       if (!default_options_.has_value()) {
         return nullptr;
       }
+      PipelineHandleT* result = Get(default_options_.value());
+      if (result != nullptr) {
+        return result;
+      }
+      SetDefault(
+          default_options_.value(),
+          std::make_unique<PipelineHandleT>(context, desc_, /*async=*/false));
```

Comment:
Ahh okay, this is where it is getting materialized and this is what `async` is doing.  Adding a comment here like `// Materialize lazy pipeline.` would be helpful.  You can even throw in a `FML_DCHECK(context.GetFlags().lazy_shader_mode)`.